pyam/utils.py

```diff
@@ -262,12 +262,18 @@ def _intuit_column_groups(df, index, include_index=False):
         existing_cols = existing_cols.union(df.columns)
 
     # check that there is no column in the timeseries data with reserved names
+    if None in existing_cols:
+        raise ValueError("Unnamed column in `data`: None")
+
+    # check that there is no column in the timeseries data with reserved/illegal names
     conflict_cols = [i for i in existing_cols if i in ILLEGAL_COLS]
     if conflict_cols:
-        msg = f"Column name {conflict_cols} is illegal for timeseries data.\n"
-        _args = ", ".join([f"{i}_1='{i}'" for i in conflict_cols])
-        msg += f"Use `IamDataFrame(..., {_args})` to rename at initialization."
-        raise ValueError(msg)
+        _cols = f"'{', '.join(conflict_cols)}'"
```

Comment:
The quotes are slightly weird, since this would become
```
'data, meta'
```

I'd take away the quotes entirely or prefer
```
'data', 'meta'
```

Reply:
Thanks for catching, I fixed that. The reason why I added the quotes is that if you have `data` and an empty string as column names, it would look like

> Illegal columns for timeseries data: , data

whereas with quotes

> Illegal columns for timeseries data: '', 'data'

(Before it was formatted as a list, not pretty).